syntax = "proto3";

package storm;

service KeyValueStore {
  rpc Get (GetRequest) returns (GetResponse);
  rpc Put (PutRequest) returns (PutResponse);
  rpc Delete (DeleteRequest) returns (DeleteResponse);
}

message GetRequest {
  string key = 1;
}

message GetResponse {
  string value = 1;
  bool success = 2;
}

message PutRequest {
  string key = 1;
  string value = 2;
}

message PutResponse {
  bool success = 1;
}

message DeleteRequest {
  string key = 1;
}

message DeleteResponse {
  bool success = 1;
}
